### DISCUSSION TITLE
Feed Direct spikes to SNN (LIF based ) in lava-dl

### DISCUSSION THREAD

**Discussion by gwgknudayanga** [lava-nc/lava-dl#309] [Q&A]
Dear community,

how can I feed the spikes from dvs camera to SNNs in laval-dl?
the collected spikes are stored as 20ms length clips. it contain 2 event polarities related to intensity increment events (positive polarity) and intensity decrement events(negative polarity) . So, can i feed events from any given 20ms clip as 10 time steps and each time step has two channels (positive polarity channel and negative polarity channel).

Thank you,
Thanks and Rgds,
Udayanga

**Comment by Michaeljurado42** [comment]
Have you checked out the Event class?

https://github.com/lava-nc/lava-dl/blob/main/src/lava/lib/dl/slayer/io.py#L12

**Comment by gwgknudayanga** [comment]
Dear Michaeljurado42,
Thank you for your reply. 

1. But that class cannot be used to feed the dvs events (that i mentioned initially in this post) to the trained SNN model that is running on Loihi 2 cloud, isn't it? For example when running pilotnet snn on Loihi 2 i can see the authors used "io.dataloader.SpikeDataloader(dataset=full_set,interval=steps_per_sample)" which is from lava.proc.io API. Also when running yolo_kp on Loihi, authors used  io.encoder.DeltaEncoder. So in my case where i have to feed dvs events what is the correponding "lava.proc.io" API that i should use and how should that should be used ?

Thank you so much.
Thanks and Rgds,
Udayanga

**Reply by Michaeljurado42** [reply]
Lava-dl is primarily used for training SNNs offline and can be used to export them to Lava via netx. This may be a question more suited for the lava repo or the INRC. (All of these API's that you listed are lava APIs!) I have still somewhat limited experience with interfacing DVS cameras directly with loihi2. Good luck!
